src/plots/polar/polar.js

```diff
@@ -1116,26 +1278,229 @@ function computeSectorBBox(sector) {
     return [x0, y0, x1, y1];
 }
 
-function pathSector(r, sector) {
-    if(isFullCircle(sector)) {
-        return Drawing.symbolFuncs[0](r);
+function isAngleInSector(rad, sector) {
+    if(isFullCircle(sector)) return true;
+
+    var s0 = wrap360(sector[0]);
+    var s1 = wrap360(sector[1]);
+    if(s0 > s1) s1 += 360;
+
+    var deg = wrap360(rad2deg(rad));
+    var nextTurnDeg = deg + 360;
+
+    return (deg >= s0 && deg <= s1) ||
+        (nextTurnDeg >= s0 && nextTurnDeg <= s1);
+}
+
+function snapToVertexAngle(a, vangles) {
+    function angleDeltaAbs(va) {
+        return Math.abs(angleDelta(a, va));
     }
 
-    var xs = r * Math.cos(deg2rad(sector[0]));
-    var ys = -r * Math.sin(deg2rad(sector[0]));
-    var xe = r * Math.cos(deg2rad(sector[1]));
-    var ye = -r * Math.sin(deg2rad(sector[1]));
+    var ind = findIndexOfMin(vangles, angleDeltaAbs);
+    return vangles[ind];
+}
 
-    var arc = Math.abs(sector[1] - sector[0]);
-    var flags = arc <= 180 ? [0, 0, 0] : [0, 1, 0];
+// taken from https://stackoverflow.com/a/2007279
+function angleDelta(a, b) {
+    var d = b - a;
+    return Math.atan2(Math.sin(d), Math.cos(d));
+}
+
+function findIndexOfMin(arr, fn) {
+    fn = fn || Lib.identity;
+
+    var min = Infinity;
+    var ind;
+
+    for(var i = 0; i < arr.length; i++) {
+        var v = fn(arr[i]);
+        if(v < min) {
+            min = v;
+            ind = i;
+        }
+    }
+    return ind;
+}
+
+// find intersection of 'v0' <-> 'v1' edge with a 'radial' line
+// (i.e. a line that starts from the origin at angle 'a')
+// given an (xp,yp) pair on the 'v0' <-> 'v1' line
+// (N.B. 'v0' and 'v1' are angles in radians)
+function findIntersectionXY(v0, v1, a, xpyp) {
+    var xstar, ystar;
+
+    var xp = xpyp[0];
+    var yp = xpyp[1];
+    var dsin = clampTiny(Math.sin(v1) - Math.sin(v0));
+    var dcos = clampTiny(Math.cos(v1) - Math.cos(v0));
+
+    if(dsin && dcos) {
+        // given
+        //  g(x) := v0 -> v1 line = m * x + b
+        //  h(x) := mr * x
+        // solve g(xstar) = h(xstar)
+        var m = dsin / dcos;
+        var b = yp - m * xp;
+        var mr = Math.sin(a) / Math.cos(a);
```

Comment:
`Math.tan(a)`

Reply:
Actually, this isn't so :laughing: , `Math.tan` isn't bounded. I didn't noticed anything wrong before because:

```js
Math.tan(Math.PI / 2)
// => 16331239353195370
```

(in Chrome 67 at least).

Oh well, https://github.com/plotly/plotly.js/pull/2739/commits/7d405a04a7298f74ef2533b2951b2a1dd72fb3ab made things more robust.